src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExperimentalOptions.cs

```diff
@@ -29,9 +29,9 @@ public ExperimentalOptions(IConfiguration configuration)
             this.EmitLogEventAttributes = emitLogEventAttributes;
         }
 
-        if (configuration.TryGetBoolValue(OpenTelemetryProtocolExporterEventSource.Log, EnableInMemoryRetryEnvVar, out var enableInMemoryRetry))
+        if (configuration.TryGetStringValue(OtlpRetryEnvVar, out var retryPolicy) && retryPolicy != null && retryPolicy.Equals("in_memory", StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
        if (configuration.TryGetStringValue(OtlpRetryEnvVar, out var retryPolicy)
            && retryPolicy.Equals("in_memory", StringComparison.OrdinalIgnoreCase))
```

You shouldn't need the `&& retryPolicy != null` part now that you are calling the `TryGetStringValue` which has the `NotNullWhen` annotation.